pkgs/shells/powershell/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv, fetchgit, dotnet-sdk, autoPatchelfHook, fetchzip, libunwind, libuuid, icu, git, curl, cmake, cacert,
+  glibcLocales, makeWrapper, less, openssl }:
+
+let bootstrap = vsn: stdenv.mkDerivation rec {
+  name = "powershell-bootstrap-${version}";
+  version = vsn;
+
+  src = fetchzip {
+    url = "https://github.com/PowerShell/PowerShell/releases/download/v${version}/powershell-${version}-linux-x64.tar.gz";
+    sha256 = "000mmv5iblnmwydfdvg5izli3vpb6l14xy4qy3smcikpf0h87fhl";
+    stripRoot = false;
+  };
+
+  buildInputs = [ autoPatchelfHook stdenv.cc.cc ];
+
+  installPhase = ''
+  mkdir -p $out/bin
+  cp -r * $out/bin
+  touch a
+  '';
+
+  dontStrip = true;
+
+
+}; in
+stdenv.mkDerivation rec {
+  name = "powershell-${version}";
+  version = "6.0.2";
+  
+  src = fetchgit {
+    url = "https://github.com/PowerShell/PowerShell.git";
+    rev = "v${version}";
+    sha256 = "0hhh33cdfs29gy73p6bp56nqqv3cr3x9r3j7ib6141arp5yn0kxf";
+    leaveDotGit = true;
+    fetchSubmodules = true;
+    deepClone = true;
+  };
+
+  buildInputs = [ dotnet-sdk autoPatchelfHook (bootstrap version) libunwind libuuid icu git curl cmake cacert glibcLocales makeWrapper ];
+  propagatedBuildInputs = [ libunwind libuuid icu curl less openssl ];
+
+  dontUseCmakeConfigure = true;
+
+  buildPhase = ''
+  # Prepare the build script
+  cat <<EOF >build.ps1
+  Import-Module ./build.psm1
+  Start-PSBootstrap
+  Start-PSBuild
+  EOF
+  # Ensure current version of the SDK is used
+  cat <<EOF >global.json
+  {
+    "sdk": {
+      "version": "`${dotnet-sdk}/bin/dotnet --version`"
+     }
+  }
+  EOF
+  # Make sure dotnet has a HOME to read/write
+  export HOME=$(mktemp -d)
+  # To prevent dotnet restore from trying to write to a read-donly filesystem
+  export DOTNET_CLI_TEST_FALLBACKFOLDER=$(mktemp -d)
+  # Disable telemetry
+  export DOTNET_CLI_TELEMETRY_OPTOUT=1
+  # Build
+  LD_LIBRARY_PATH="${stdenv.lib.makeLibraryPath [ libunwind libuuid icu ]}:$LD_LIBRARY_PATH" ${(bootstrap version)}/bin/pwsh build.ps1
+  '';
+  
+  installPhase = ''
+  mkdir -p $out/bin
+  cp -r src/powershell-unix/bin/Linux/netcoreapp2.0/linux-x64/publish/* $out
+  rm $out/DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY
+  makeWrapper $out/pwsh $out/bin/pwsh --prefix LD_LIBRARY_PATH : "${stdenv.lib.makeLibraryPath [ libunwind libuuid icu openssl curl ]}" \
+                                      --set PAGER ${less}/bin/less --set TERM linux
+  '';
+
+  dontStrip = true;
+
+  meta = with stdenv.lib; {
+    description = "PowerShell Core";
```

Comment:
The description should not contain the name. C.f. https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes